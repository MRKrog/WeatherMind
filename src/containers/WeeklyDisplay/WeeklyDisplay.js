import React, { Component } from 'react';
import { connect } from 'react-redux';
import { WeekDetail } from '../../components/WeekDetail/WeekDetail'

export class WeeklyDisplay extends Component {

    render() {
      const { weekly } = this.props;

      return (
        <div className="WeeklyDisplay">
          {
            weekly.map(week => {
              return <WeekDetail key={week.time} {...week} />
            })
          }
        </div>
      )
    }
}

export const mapStateToProps = (state) => ({
    weekly: state.weekly
})

export default connect(mapStateToProps)(WeeklyDisplay)
