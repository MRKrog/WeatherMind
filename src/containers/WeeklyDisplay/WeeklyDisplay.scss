.WeeklyDisplay {
  display: flex;
  flex-direction: column;
  background: #1e182b;
  padding: 1em;
  color: #fff;
}
