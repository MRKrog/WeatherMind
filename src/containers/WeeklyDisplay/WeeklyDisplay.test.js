import React, { Component } from 'react';
import { WeeklyDisplay, mapStateToProps } from './WeeklyDisplay';
import { shallow } from 'enzyme';

const mockWeeklyData = [
    { icon: "partly-cloudy-day", tempHigh: 74, tempLow: 44, time: "Monday" },
    { icon: "partly-cloudy-day", tempHigh: 71, tempLow: 44, time: "Tuesday" },
    { icon: "partly-cloudy-day", tempHigh: 58, tempLow: 37, time: "Wednesday" },
    { icon: "snow", tempHigh: 60, tempLow: 38, time: "Thursday" },
    { icon: "clear-day", tempHigh: 73, tempLow: 45, time: "Friday" },
    { icon: "partly-cloudy-night", tempHigh: 78, tempLow: 48, time: "Saturday" },
    { icon: "rain", tempHigh: 61, tempLow: 41, time: "Sunday" },
  ]


describe('WeeklyDisplay', () => {

  describe('WeeklyDisplay Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<WeeklyDisplay weekly={mockWeeklyData} />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

  })

  describe('mapStateToProps', () => {

    it('should return an object with the today arrayâ€™', () => {

    })

  })

})
