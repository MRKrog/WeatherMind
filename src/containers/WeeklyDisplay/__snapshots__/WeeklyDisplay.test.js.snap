// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyDisplay WeeklyDisplay Component should match the snapshot 1`] = `
<div
  className="WeeklyDisplay"
>
  <WeekDetail
    icon="partly-cloudy-day"
    key="Monday"
    tempHigh={74}
    tempLow={44}
    time="Monday"
  />
  <WeekDetail
    icon="partly-cloudy-day"
    key="Tuesday"
    tempHigh={71}
    tempLow={44}
    time="Tuesday"
  />
  <WeekDetail
    icon="partly-cloudy-day"
    key="Wednesday"
    tempHigh={58}
    tempLow={37}
    time="Wednesday"
  />
  <WeekDetail
    icon="snow"
    key="Thursday"
    tempHigh={60}
    tempLow={38}
    time="Thursday"
  />
  <WeekDetail
    icon="clear-day"
    key="Friday"
    tempHigh={73}
    tempLow={45}
    time="Friday"
  />
  <WeekDetail
    icon="partly-cloudy-night"
    key="Saturday"
    tempHigh={78}
    tempLow={48}
    time="Saturday"
  />
  <WeekDetail
    icon="rain"
    key="Sunday"
    tempHigh={61}
    tempLow={41}
    time="Sunday"
  />
</div>
`;
